UpdatePublishedVersions.ps1

```diff
@@ -0,0 +1,149 @@
+﻿#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+# This script updates the dotnet/versions repository based on a set of packages. It directly
+# commits the changes using GitHub APIs.
+
+param(
+    [Parameter(Mandatory=$true)][string]$gitHubUser,
+    [Parameter(Mandatory=$true)][string]$gitHubEmail,
+    [Parameter(Mandatory=$true)][string]$gitHubAuthToken,
+    [Parameter(Mandatory=$true)][string]$versionsRepoOwner,
+    [Parameter(Mandatory=$true)][string]$versionsRepo,
+    [Parameter(Mandatory=$true)][string]$versionsRepoPath,
+    # A pattern matching all packages in the set that the versions repository should be set to.
+    [Parameter(Mandatory=$true)][string]$nupkgPath,
+    # Print out the new file contents, but don't change the versions repository.
+    [switch]$dryRun)
+
+function ConvertPathTo-Package([string]$path)
+{
+    # Find the package ID and version using a regex. This matches the semantic version
+    # and assumes everything to the left is the id or a path to the package directory.
+    $matched = $path -match '^(.*\\)?(.*?)\.(([0-9]+\.)?[0-9]+\.[0-9]+(-([A-z0-9-]+))?)\.(symbols\.)?nupkg$'
+    if ($matched)
+    {
+        $packageInfo = @{
+            Path = $path
+            Name = $matches[2]
+            Version = $matches[3]
+            Prerelease = $matches[6]
+        }
+        $packageInfo.NameVersion = "$($packageInfo.Name) $($packageInfo.Version)"
+        return $packageInfo
+    }
+    else
+    {
+        throw "Couldn't find name and version from path $path."
+    }
+}
+
+# Updates a GitHub file with the specified file contents
+function Update-GitHub-File(
+    [string]$user = $gitHubUser,
+    [string]$email = $gitHubEmail,
+    [string]$authToken = $gitHubAuthToken,
+    [string]$owner = $versionsRepoOwner,
+    [string]$repo = $versionsRepo,
+    [string]$path,
+    [string]$newFileContent,
+    [string]$commitMessage)
+{
+    function message([string]$message)
+    {
+        Write-Host -ForegroundColor Green "*** $message ***"
+    }
+
+    $headers = @{
+        'Accept' = 'application/vnd.github.v3+json'
+        'Authorization' = "token $authToken"
+    }
+
+    $fileUrl = "https://api.github.com/repos/$owner/$repo/contents/$path"
+
+    message "Getting the `"sha`" of the current contents of file '$owner/$repo/$path'"
+
+    $currentFile = Invoke-WebRequest $fileUrl -UseBasicParsing -Headers $headers
+    $currentSha = (ConvertFrom-Json $currentFile.Content).sha
+
+    message "Got `"sha`" value of '$currentSha'"
+
+    # Base64 encode the string
+    function ToBase64([string]$value)
```

Comment:
It might make more sense to stick this method directly before it is used (i.e. down a few lines).